packages/browser/src/request.ts

```diff
@@ -68,19 +86,21 @@ const encodeToDataString = (data: string | Record<string, any>): string => {
     return 'data=' + encodeURIComponent(typeof data === 'string' ? data : jsonStringify(data))
 }
 
-const encodePostData = ({ data, compression }: RequestWithOptions): EncodedBody | undefined => {
+const encodePostData = async ({ data, compression }: RequestWithOptions): Promise<EncodedBody | undefined> => {
     if (!data) {
         return
     }
 
     if (compression === Compression.GZipJS) {
```

Comment:
We should probably document somewhere that if you declare `GZipJS` compression we might NOT compress it in the rare case where the browser doesn't provide a `CompressionStream` interface